### PR TITLE
APM Server management modes

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR:

#### Additions

- [x] Adds initial content changes for the new `Quick start with Elastic Cloud` and `Get started (self-managed)` sections. More work still needs to be done here.

#### Removals
- [x] Removes the `Legacy APM Overview` ([this](https://www.elastic.co/guide/en/apm/guide/current/legacy-apm-overview.html)).
- [x] Deletes the duplicate `API` section in the APM Server Reference ([this](https://www.elastic.co/guide/en/apm/guide/current/intake-api.html))
- [x] Deletes the duplicate `How to` section in the APM Server Reference ([this](https://www.elastic.co/guide/en/apm/guide/current/howto-guides.html))
- [x] Removes all deprecation notices from the `Legacy APM Overview` and the `Legacy APM Server Reference`.

#### Changes
- [x] Up-levels the remaining pages in the `Legacy APM Server Reference`. This will make it easier to combine the content with the APM Guide. You can see this change in the [TOC in the doc preview](https://apm-server_10560.docs-preview.app.elstc.co/guide/en/apm/guide/master/getting-started-apm-server.html).

### CODE REVIEW

File: docs/images/decision-tree.png
Comment (simitt) [inline docs/images/decision-tree.png:1]: let's discuss the decision tree live, but here some first thoughts (and I'm aware that this is largely based on my draft, so basically finding issues with my own suggestion here ;) ):
- Where will you run APM Server? The current wording implies that it should be run on edge for backend agents. I would not go that far. We still optimized for central apm-server. Maybe rephrase towards, on edge is possible for non-RUM agents. 
- If we generally describe _Standalone Elastic Agent managed APM Server_ then we should also add it to this diagram: 
  - Do you want to use other integrations? 
    - -> yes: Would you like to use the comfort of Fleet UI based management? -> yes: Fleet managed APM Server; -> no: Standalone Elastic Agent managed APM Server
    - -> no: What is your prefered way of configuration?  -> yaml: APM Server binary; -> Kibana UI: Fleet managed APM Server

File: docs/apm-components.asciidoc [hunk 1]
```diff
@@ -1,39 +1,150 @@
...
+
+How?
+
+1. Install APM Server
+2. Configure APM Server
+3. Run APM Server
+
+**Pros**:
+
+- Automation?
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/apm-components.asciidoc:62]: + remove cons

File: docs/apm-components.asciidoc [hunk 2]
```diff
@@ -1,39 +1,150 @@
...
+[float]
+===== Fleet-managed APM Server
+Simplify the configuration and management of your APM Server instances by using Fleet to manage APM Server.
+Fleet is a web-based UI in Kibana for centrally managing {agent}s.
+Instantly roll out configuration changes and upgrades across your infrastructure.
+
+How?
+
+1. Install Elastic Agent on your edge machines
+2. Add a Fleet Server
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/apm-components.asciidoc:120]: Super convenient config. Super easy to add other integrations.

File: docs/apm-components.asciidoc [hunk 3]
```diff
@@ -1,39 +1,150 @@
...
-// Not sure which to choose? See the [blog post]
+Components required by method:
+
+[options="header"]
+|====
+| Installation method                         | APM Server | Elastic Agent | Fleet Server
+| APM Server binary                           | ✔️          |               |
+| Standalone Elastic Agent-managed APM Server | ✔️          | ✔️             |
+| Fleet-managed APM Server                    | ✔️          | ✔️             | ✔️
+|====
```
Comment (bmorelli25) [inline docs/apm-components.asciidoc:48]: move integration to the Elastic Stack

**Comment by bmorelli25** [comment]
Thanks for the feedback, @simitt. I moved the new content out of the **Components and documentation** page and into two new top-level nav headings:

Screenshot 2023-04-05 at 5 03 14 PM

Here are preview links to each of these pages:

* [Quick start with Elastic Cloud](https://apm-server_10560.docs-preview.app.elstc.co/guide/en/apm/guide/master/apm-quickstart-new.html)
* [Get started (self-managed)](https://apm-server_10560.docs-preview.app.elstc.co/guide/en/apm/guide/master/apm-setup-new.html)

File: docs/_quickstart.asciidoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+* Are your eyes bleary from staring at a wall of screens?
+{observability-guide}/create-alerts.html[Create alerts] and find out about
+problems while sipping your favorite beverage poolside.
+
+* Want Elastic to do the heavy lifting? Use {ml} to
+{observability-guide}/inspect-log-anomalies.html[detect anomalies].
+
+* Got everything working like you want it? Roll out your agent policies to
+other hosts by deploying {agent}s across your infrastructure!
```
Comment (simitt) [inline docs/_quickstart.asciidoc:100]: Should we remove this? I'd like to have as little confusion with APM agents & Elastic Agent on this page as possible.

File: docs/_get-started.asciidoc [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+- {es}
+- {ess}
+
+**Required components**:
+
+- APM agents
+- APM Server
+- {agent}
+- {fleet}
```
Comment (simitt) [inline docs/_get-started.asciidoc:80]: should rather be `Fleet Server` (Fleet is part of Kibana, so will always be there, but Fleet Server needs to be run as a component

**Review by simitt — COMMENTED** [review]
This looks great!! Very excited about these changes.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
This PR has been replaced by https://github.com/elastic/apm-server/pull/10750 and https://github.com/elastic/apm-server/pull/10751
